packages/react-instantsearch-core-next/src/useIndex.ts

```diff
@@ -0,0 +1,23 @@
+import index from 'instantsearch.js/es/widgets/index/index';
+import { useEffect, useMemo } from 'react';
+
+import { useIndexContext } from './useIndexContext';
+
+import type { IndexWidgetParams } from 'instantsearch.js/es/widgets/index/index';
+
+export type UseIndexProps = IndexWidgetParams;
+
+export function useIndex(props: UseIndexProps) {
+  const searchIndex = useIndexContext();
+  const indexWidget = useMemo(() => index(props), [props]);
```

Comment:
should we leave a comment that the memo is using a non-stable parameter? Are we fixing this somewhere consistently?

Reply:
I already addressed this in the PR that's coming.